### DISCUSSION TITLE
Using QLoRA to fine-tune BaiChuan-13B reports AttributeError: 'Parameter' object has no attribute 'weight'.

### DISCUSSION THREAD

**Discussion by Aitejiu** [huggingface/peft#1179] [Q&A]
`package version:peft==0.6.2`
Using `QLoRA` to fine-tune `BaiChuan-13B` reports `AttributeError: 'Parameter' object has no attribute 'weight'`.
The error location code is
![image](https://github.com/huggingface/peft/assets/73458598/b140a873-26bd-45af-9ba7-e3c47182d12f)
**Its in /peft/tuners/adalora/bnb.py line 144.**
After looking through the source code, I found that `lora_A[active_adapter]` is considered a **nn.module** in `peft==0.6.2` because it has the weight attribute. But after the model is loaded, `lora_A[active_adapter]` is actually an `nn.parameters`, it is already the **weight of the corresponding module**. I would like to know why this is the case?

**Comment by younesbelkada** [comment]
Hi @Aitejiu , I believe It has been fixed with https://github.com/huggingface/peft/pull/1146 can you please double check by installing PEFT from source? `pip install -U git+https://github.com/huggingface/peft.git`
